test/performance/shmem_perf_suite/common.h

```diff
@@ -0,0 +1,83 @@
+/*
+*
+*  Copyright (c) 2015 Intel Corporation. All rights reserved.
+*  This software is available to you under the BSD license. For
+*  license information, see the LICENSE file in the top level directory.
+*
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <shmem.h>
+#include <shmemx.h>
+#include <string.h>
+#include <assert.h>
+
+char * aligned_buffer_alloc(int len) {
+    unsigned long page_align;
+    char *buf;
+
+    page_align = getpagesize();
+    buf = shmem_malloc(len + page_align);
+    buf = (char *) (((unsigned long) buf + (page_align - 1)) /
+            page_align * page_align);
+
+    return buf;
+}
+
+int static inline is_divisible_by_4(int num)
+{
+    assert(sizeof(int) == 4);
+    return (!(num & 0x00000003));
+}
+
+/*to be a power of 2 must only have 1 set bit*/
+int static inline is_pow_of_2(unsigned int num)
+{
+    int i = 0;
+    unsigned int count = 0;
+
+    do {
+        if(num & 1)
+            count++;
```

Comment:
I think that you're gathering the population count of ```num``` into ```count```, which is more than you need.  You should be able to replace ```count++``` with ```return num == 1``` and drop the next ```if``` block.

I realize that this code section is likely not on a performance sensitive path, but I can't help myself when it comes to bit twiddling.  :)